doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -877,30 +900,30 @@ if it reduces the subterm under consideration. The ``fold`` strategy takes
 a :token:`term` and tries to *unify* it to the current subterm, converting it to :token:`term`
 on success. It is stronger than the tactic ``fold``.
 
+.. _rewrite_strat_innermost_outermost:
 
-Usage
-~~~~~
+.. example:: :n:`innermost` and :n:`outermost`
 
+   The type of `Nat.add_comm` is `forall n m : nat, n + m = m + n`.
 
-.. tacn:: rewrite_strat @rewstrategy {? in @ident }
-   :name: rewrite_strat
+   .. coqtop:: all
 
-   Rewrite using the strategy s in hypothesis ident or the conclusion.
+      Require Import Coq.Arith.Arith.
```

Comment:
Do you not also need `Require Import Coq.Setoids.Setoid.`?

Reply:
No.